nautobot/extras/management/__init__.py

```diff
@@ -153,12 +229,21 @@ def create_custom_statuses(
         # Do not print output during unit testing migrations
         verbosity = 1
 
-    if not models:
-        models = CHOICESET_MAP.keys()
-
-    # Prep the app and get the Status model dynamically
+    choiceset_map = {}
+    if metadata_model.lower() == "role":
+        choiceset_map = ROLE_CHOICESET_MAP
+        if not models:
+            models = choiceset_map.keys()
+    elif metadata_model.lower() == "status":
+        choiceset_map = STATUS_CHOICESET_MAP
+        if not models:
+            models = choiceset_map.keys()
+    else:
+        raise ValueError("only status and role are supported for _create_custom_role_or_status_instances")
+
+    # Prep the app and get the model dynamically
     try:
-        Status = apps.get_model("extras.Status")
+        Metadata_Model = apps.get_model(f"extras.{metadata_model}")
```

Comment:
Let's not mix `CamelCase` and `snake_case` - mind changing this to `MetadataModel` throughout?